WeWatch/WeWatch/Splash/SplashViewModel.swift

```diff
@@ -0,0 +1,106 @@
+////
+////  SplashView.swift
+////  WeWatch
+////
+////  Created by Anton on 18/12/2024.
+////
+//
+import Foundation
+
+@MainActor
+class SplashViewModel: ObservableObject {
+    
+    internal enum viewJwtError: Error {
+        case typeChangeError
+        case invalidExpirationClaim
+        case dataError
+    }
+    
+    @Published internal var isLoading: Bool = false
+    @Published internal var token: String? = nil
+    @Published internal var errorMessage: String? = nil
+    @Published var showMainView: Bool = false
+    
+    func loginToSplashView() async {
+        if isValidToken() {
+            self.showMainView = true
+        } else {
+            guard let loginData: Data = prepareLoginRequest() else {
+                self.errorMessage = "failed to encode login data."
+                return
+            }
+            let loginResource: Resource<LoginResponse> = .init(
+                url: URL.loginURL,
+                method: .post(loginData)
+            )
+            do {
+                let response: LoginResponse = try await Webservice().call(loginResource)
+                if let token: String = response.data?.token {
+                    self.token = token
+                    do {
+                        try KeychainManager.store(data: token, key: "token")
+                    } catch {
+                        self.errorMessage = "Failed to store token: \(error.localizedDescription)"
+                    }
+                } else {
+                    self.errorMessage = "Failed to retriev token"
+                }
+            } catch {
+                self.errorMessage = "Error during login: \(error.localizedDescription)"
+            }
+        }
+    }
+    
+    internal func decodingJwtToken() throws -> [String: Any] {
+        guard let tokenData: Data = try? KeychainManager.getData(key: "token") else {
```

Comment:
If your function throws, there's no point in using `try?` because your function will do it anyway.